lib/ohai/dsl/plugin.rb

```diff
@@ -183,6 +185,15 @@ def method_missing(name, *args)
 
       private
 
+      def safe_get_attribute(*keys)
+        attrs = @data
+        keys.each do |key|
+          return nil unless attrs.respond_to?(:has_key?) && attrs.has_key?(key)
```

Comment:
copied and adapted from some other code of mine (still a bit Before Caffeine for me so I may be overlooking something):

```ruby
begin
  keys.inject(@data) { |memo, key| memo[key] }
rescue NoMethodError
  nil
end
```